climada/util/plot.py

```diff
@@ -736,31 +737,37 @@ def get_transformation(crs_in):
     crs_epsg : ccrs.Projection
     units : str
     """
-    try:
-        if CRS.from_user_input(crs_in) == CRS.from_user_input('EPSG:3395'):
-            crs_epsg = ccrs.Mercator()
-        else:
-            crs_epsg = ccrs.epsg(CRS.from_user_input(crs_in).to_epsg())
-    except ValueError:
-        crs_epsg = ccrs.PlateCarree()
-    except requests.exceptions.ConnectionError:
-        LOGGER.warning('No internet connection.'
-                       ' Using projection PlateCarree in plot.')
-        crs_epsg = ccrs.PlateCarree()
+    with warnings.catch_warnings():
+    # Within this function we want to suppress the UserWarning from pyproj._crs._CRS.to_proj4:
+    # You will likely lose important projection information when converting to a PROJ string from
+    # another format. See:
+    # https://proj.org/faq.html#what-is-the-best-format-for-describing-coordinate-reference-systems
+        warnings.simplefilter(action="ignore", category=UserWarning)
 
-    try:
-        units = (crs_epsg.proj4_params.get('units')
+        try:
+            if CRS.from_user_input(crs_in) == CRS.from_user_input('EPSG:3395'):
+                crs_epsg = ccrs.Mercator()
+            else:
+                crs_epsg = ccrs.epsg(CRS.from_user_input(crs_in).to_epsg())
+        except ValueError:
+            crs_epsg = ccrs.PlateCarree()
+        except requests.exceptions.ConnectionError:
+            LOGGER.warning('No internet connection. Using projection PlateCarree in plot.')
+            crs_epsg = ccrs.PlateCarree()
```

Comment:
I suggest setting the overall fallback value first (this avoids setting it twice in differenct `catch` statements), and adding a warning in case of a parsing error:
```suggestion
        crs_epsg = ccrs.PlateCarree()  # Fallback value
        try:
            if CRS.from_user_input(crs_in) == CRS.from_user_input('EPSG:3395'):
                crs_epsg = ccrs.Mercator()
            else:
                crs_epsg = ccrs.epsg(CRS.from_user_input(crs_in).to_epsg())
        except ValueError:
            LOGGER.warning(f"Error parsing coordinate system '{crs_in}'."
                "Using projection PlateCarree in plot.")
        except requests.exceptions.ConnectionError:
            LOGGER.warning('No internet connection. Using projection PlateCarree in plot.')
```

Reply:
Thanks for the suggestions!

I agree to the warning, but to the prior fallback value setting I'm actually quite opposed. In general it does not help with readability and it does not help with avoiding mistakes. The only advantage I can see is that it saves some lines of code. While in the mean time code will be executed unnecessarily.

Reply:
Fair enough. I wanted to avoid doubled lines of code but I see your point. I'll resolve this conversation and commit the warning separately. Then I'll merge ✌️